Revision of the occ maintenance command description (#189)
owncloud/docs-server: modules/admin_manual/pages/configuration/server/occ_commands/core_commands/_maintenance_commands.adoc

 maintenance:update:htaccess         Updates the .htaccess file
----

== Update the Systems Data-Fingerprint

> When a backup has been restored, the ETag information which is necessary when accessing ownCloud with clients has been changed. Run following command to tell desktop and mobile clients that a server backup has been restored.

Reviewer: ```suggestion
When a backup has been restored, the ETag information, which is necessary when accessing ownCloud with clients, has been changed. Run the following command to tell desktop and mobile clients that a server backup has been restored.
```

[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} maintenance:data-fingerprint
----

This command changes the ETag for all files in the communication with sync clients, informing them that one or more files were modified. After the command completes, users will be prompted to resolve any conflicts between newer and older file versions.

> == Install owncloud

Reviewer: ```suggestion
== Install ownCloud
```

> NOTE: This command is only available if the following key in your `config.php` is not present or set to `false`. This is the case when the installation has not been made or not finalized. This key automatically turns to `true` after a successful installation resulting that the command is not longer available.

Reviewer: ```suggestion
NOTE: This command is only available if the following key in your `config.php` is not present or set to `false`. This is the case when the installation has not been made or not finalized. This key automatically turns to `true` after a successful installation. This results in the command no longer being available.
```

[source,php]
----
'installed' => false,
----

[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} maintenance:install [options]
----

The `maintenance:install` command supports the following options:

[cols="30%,75%",options="header"]
|===
| Option 
| Description

| `--database`=DATABASE
| Supported database type. [default: `sqlite`] +
Possible values: `sqlite` ,`mysql`, `pgsql`, `oci` +

> Note that `oci` (Oracle) is only available with the Enterprise license 

Reviewer: ```suggestion
Note that `oci` (Oracle) is only available with the Enterprise license.
```

| `--database-connection-string`=DATABASE-CONNECTION-STRING

> | Oracle specific connection string. As soon as this parameter is provided other parameters like database-host and database-name are not used and do not need to be provided

Reviewer: ```suggestion
| Oracle specific connection string. As soon as this parameter is provided, other parameters like database-host and database-name are not used and do not need to be provided.
```

| Hostname of the database. [default: "localhost"]

| `--database-user`=DATABASE-USER
| User name to connect to the database.

| `--database-pass`=DATABASE-PASS
| Password of the database user.

| `--database-table-prefix`=DATABASE-TABLE-PREFIX
| Prefix for all tables (default: oc_).

| `--admin-user`=ADMIN-USER
| User name of the admin account. [default: "admin"]

| `--admin-pass`=ADMIN-PASS
|  Password of the admin account.

| `--data-dir`=DATA-DIR
| Path to the data directory. [default: "/var/www/owncloud/data"]
|===

== Update Database Mimetypes

Update database mimetypes and file cache.

Usage:

[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} maintenance:mimetype:update-db [options]
----

The `maintenance:mimetype:update-db` command supports the following options:

[cols="25%,75%",options="header"]
|===
| Option 
| Description

| `--repair-filecache`
| Repair the file cache for all mimetypes, not just the new ones.
|===

== Update the mimetypelist.js File

This command updates the `mimetypelist.js` file.

Usage:

[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} maintenance:update-js
----

== Enable or Disable Maintenance Mode

> `maintenance:mode` command locks the sessions of all logged-in users, including administrators, and displays a status screen warning that the server is in maintenance mode. Users who are not already logged in cannot log in until maintenance mode is turned off. When you take the server out of maintenance mode logged-in users must refresh their Web browsers to continue working.

Reviewer: ```suggestion
`maintenance:mode` command locks the sessions of all logged-in users, including administrators, and displays a status screen warning that the server is in maintenance mode. Users who are not already logged in cannot log in until maintenance mode is turned off. Once you take the server out of maintenance mode, logged-in users must refresh their Web browsers to continue working.
```

----
{occ-command-example-prefix} maintenance:mode [options]
----

The `maintenance:repair` command supports the following options:

[cols="25%,75%",options="header"]
|===
| Option 
| Description

| `--on`
| Enable maintenance mode.

| `--off`
| Disable maintenance mode.
|===

Turn on maintenance mode:

[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} maintenance:mode --on
----

Turn it off when you’re finished with the maintenance tasks:

[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} maintenance:mode --off
----

== Installation Repair Commands

The `maintenance:repair` command helps administrators repair an installation. The command runs automatically during upgrades to clean up the database. So, while you can run it manually, there usually isn't a need to.

NOTE: Your ownCloud installation needs to be in maintenance mode to use the `maintenance:repair` command.

Usage:

[source,bash,subs="attributes+"]
----
{occ-command-example-prefix} maintenance:repair [options]
----

=== Repair Command Options

The `maintenance:repair` command supports the following options:

[cols="25%,75%",options="header"]
|===
| Option 
| Description

| `--list`

> | Lists all possible repair steps

Reviewer: ```suggestion
| Lists all possible repair steps.
```
